### PR TITLE
Groom `PartialCellBottom` for prime time

### LINKED ISSUE #3064: Should reductions exclude `peripheral_node`?
Right now reductions only exclude _immersed_ peripheral nodes:

https://github.com/CliMA/Oceananigans.jl/blob/72e2197e640ddf308fd40aa4658e7686e596bccd/src/ImmersedBoundaries/immersed_reductions.jl#L27

It recently caused me a lot of pain and confusion and time that "ordinary" peripheral nodes are included in the reduction, but immersed peripheral nodes are not.

What is the logic for treating immersed boundaries differently from ordinary boundaries? I think we should either exclude only inactive nodes or peripheral nodes, but this behavior should be consistent between immersed and not immersed grids.

This change would mean we don't need special reductions (at least for fields that are not reduced) on immersed vs not-immersed grids.

@simone-silvestri may have the answer.

### PR DESCRIPTION
**Pull request by glwagner**
This PR readies `PartialCellBottom` for prime time. Mostly we reorganize code; the substantial changes are:

* Use `Field{Center, Center, Nothing}` for bottom_height. This is convenient mostly for `show` / `summary` but may also prove helpful for users.
* Extend implicit vertical diffusion functionality for partial cells. I'm not sure we did this right...

Closes #3064

### CODE REVIEW

**Comment by navidcy** [comment]
some tests require attention :)

File: Manifest.toml
Comment (navidcy) [inline Manifest.toml:1]: somehow manifest was deleted?

File: src/Fields/field.jl [hunk 1]
```diff
@@ -545,12 +545,12 @@ const MinimumReduction = typeof(Base.minimum!)
 const AllReduction     = typeof(Base.all!)
 const AnyReduction     = typeof(Base.any!)
 
-check_version_larger_than_7() = VERSION.minor > 7
+isversion8⁺() = VERSION.minor > 7
```
Comment (navidcy) [inline src/Fields/field.jl:548]:
```suggestion
isjuliaversion1p8⁺() = VERSION.minor > 7
```

File: Manifest.toml
Comment (glwagner) [reply Manifest.toml:1]: i'll put it back

File: src/Fields/field.jl [hunk 2]
```diff
@@ -545,12 +545,12 @@ const MinimumReduction = typeof(Base.minimum!)
...
-initialize_reduced_field!(::AnyReduction,  f, r::ReducedField, c) = check_version_larger_than_7() ? Base.initarray!(interior(r), f, |, true, interior(c))             : Base.initarray!(interior(r), |, true, interior(c))
+initialize_reduced_field!(::SumReduction,  f, r::ReducedField, c) = isversion8⁺() ? Base.initarray!(interior(r), f, Base.add_sum, true, interior(c))  : Base.initarray!(interior(r), Base.add_sum, true, interior(c))
+initialize_reduced_field!(::ProdReduction, f, r::ReducedField, c) = isversion8⁺() ? Base.initarray!(interior(r), f, Base.mul_prod, true, interior(c)) : Base.initarray!(interior(r), Base.mul_prod, true, interior(c))
+initialize_reduced_field!(::AllReduction,  f, r::ReducedField, c) = isversion8⁺() ? Base.initarray!(interior(r), f, &, true, interior(c))             : Base.initarray!(interior(r), &, true, interior(c))
+initialize_reduced_field!(::AnyReduction,  f, r::ReducedField, c) = isversion8⁺() ? Base.initarray!(interior(r), f, |, true, interior(c))             : Base.initarray!(interior(r), |, true, interior(c))
```
Comment (glwagner) [inline src/Fields/field.jl:553]:
```suggestion
initialize_reduced_field!(::SumReduction,  f, r::ReducedField, c) = isjuliaversion1p8⁺() ? Base.initarray!(interior(r), f, Base.add_sum, true, interior(c))  : Base.initarray!(interior(r), Base.add_sum, true, interior(c))
initialize_reduced_field!(::ProdReduction, f, r::ReducedField, c) = isjuliaversion1p8⁺() ? Base.initarray!(interior(r), f, Base.mul_prod, true, interior(c)) : Base.initarray!(interior(r), Base.mul_prod, true, interior(c))
initialize_reduced_field!(::AllReduction,  f, r::ReducedField, c) = isjuliaversion1p8⁺() ? Base.initarray!(interior(r), f, &, true, interior(c))             : Base.initarray!(interior(r), &, true, interior(c))
initialize_reduced_field!(::AnyReduction,  f, r::ReducedField, c) = isjuliaversion1p8⁺() ? Base.initarray!(interior(r), f, |, true, interior(c))             : Base.initarray!(interior(r), |, true, interior(c))
```

**Comment by glwagner** [comment]
Superceded by #3090

**Pull request closed without merging**